x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportStopTrainedModelDeploymentAction.java

```diff
@@ -101,7 +102,17 @@ protected void doExecute(Task task, StopTrainedModelDeploymentAction.Request req
                     listener.onResponse(new StopTrainedModelDeploymentAction.Response(true));
                     return;
                 }
-                normalUndeploy(task, models.get(0).getModelId(), maybeAllocation.get(), request, listener);
+                final String modelId = models.get(0).getModelId();
+                trainedModelAllocationService.stopModelAllocation(modelId, ActionListener.wrap(
```

Comment:
Instead of adding a whole new action for this, would it be an option to call the update allocation action and update its state to stopping through that?

Reply:
> would it be an option to call the update allocation action and update its state to stopping through that?

Maybe, but there is no "update allocation" action. The only updates that occur throughout the life time of the allocation are route updates.

Reply:
This is the one I meant `UpdateTrainedModelAllocationStateAction`

Reply:
> UpdateTrainedModelAllocationStateAction

The focus of that request is only routes. Possibly I should update it, but right now that action is only requested by the node service to update a route in the cluster service.

Reply:
I updated the deployment stop action to update the allocation state to `stopping`